services/airline_service.go

```diff
@@ -2,29 +2,51 @@ package services
 
 import (
 	"airport-app-backend/models"
+	"errors"
+	"regexp"
 )
 
 type IAirlineRepository interface {
-	GetAirline(pageNum int) ([]models.Airlines, error)
-	GetAirlineById(string) (*models.Airlines, error)
+	GetAirline(int) ([]models.Airline, error)
+	GetAirlineById(string) (*models.Airline, error)
+	CreateNewAirline(*models.Airline) error
 }
 
 var DEFAULT_PAGE_LIMIT int = 10
 
-func (sr *ServiceRepository) GetAirline(pageNum int) ([]models.Airlines, error) {
-	var airlines []models.Airlines
-	result := sr.db.Limit(DEFAULT_PAGE_LIMIT).Offset(pageNum * DEFAULT_PAGE_LIMIT).Find(&airlines)
+
+func (sr *ServiceRepository) GetAirline(pageNum int) ([]models.Airline, error) {
```

Comment:
What happens if this controller action method is called without a page number? It should be an optional parameter